channel/src/main/java/com/synopsys/integration/alert/channel/email2/util/EmailAddressGatherer.java

```diff
@@ -0,0 +1,124 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.email2.util;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.exception.AlertConfigurationException;
+import com.synopsys.integration.alert.common.persistence.accessor.JobAccessor;
+import com.synopsys.integration.alert.common.persistence.accessor.ProviderDataAccessor;
+import com.synopsys.integration.alert.common.persistence.model.ProviderProject;
+import com.synopsys.integration.alert.common.persistence.model.ProviderUserModel;
+import com.synopsys.integration.alert.common.persistence.model.job.DistributionJobModel;
+import com.synopsys.integration.alert.common.persistence.model.job.details.EmailJobDetailsModel;
+
+@Component
+public class EmailAddressGatherer {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final JobAccessor jobAccessor;
+    private final ProviderDataAccessor providerDataAccessor;
+
+    @Autowired
+    public EmailAddressGatherer(JobAccessor jobAccessor, ProviderDataAccessor providerDataAccessor) {
+        this.jobAccessor = jobAccessor;
+        this.providerDataAccessor = providerDataAccessor;
+    }
+
+    public Set<String> gatherEmailAddresses(EmailJobDetailsModel emailJobDetails, Collection<String> projectHrefs) {
+        Set<String> emailAddresses = new HashSet<>();
+
+        emailAddresses.addAll(emailJobDetails.getAdditionalEmailAddresses());
+        if (emailJobDetails.isAdditionalEmailAddressesOnly()) {
+            return emailAddresses;
+        }
+
+        Optional<Long> optionalBlackDuckGlobalConfigId = jobAccessor.getJobById(emailJobDetails.getJobId())
+                                                             .map(DistributionJobModel::getBlackDuckGlobalConfigId);
+        if (optionalBlackDuckGlobalConfigId.isPresent()) {
+            Set<String> providerEmailAddresses = gatherProviderEmailAddresses(emailJobDetails.isProjectOwnerOnly(), projectHrefs, optionalBlackDuckGlobalConfigId.get());
+            emailAddresses.addAll(providerEmailAddresses);
```

Comment:
According to the description of the Project Owner Only field, this does not seem correct. I think this is a reasonable change but the Project Owner Only field description should be updated to mention the additional email addresses will be included as well. We should also add a release note and make sure the documentation is updated to match.

Reply:
Our API constrains it so that these cannot both be true, but I think you raising this point indicates that we shouldn't make that assumption here. I don't think we need to change the feature, I'll just update the logic to be more explicit about mutual exclusion.